adlfs/spec.py

```diff
@@ -1169,7 +1174,7 @@ def put(self, lpath, rpath, recursive=False, **kwargs):
         rpaths = other_paths(lpaths, rpath)
 
         for lpath, rpath in zip(lpaths, rpaths):
-            maybe_sync(self.put_file, self, lpath, rpath, **kwargs)
+            maybe_sync(self._put_file, self, lpath, rpath, **kwargs)
```

Comment:
The superclass method does this well no?
If not, recommend you still make an `async def _put`, and in there use `asyncio.gather`, so that you proprecc concurrently and get the possible speedup.